### ISSUE TITLE
Add `option.onClose` for `openModal` fn.

### ISSUE THREAD

**Issue by michaltaberski** [mattjennings/svelte-modals#27]
I think adding a close handler to `openModal` fn could make the componet very flexible.

Imagine a scenario like this:

```ts
const confirmModal = ({question}) => new Promise(resolve => 
  openModal(ConfirmModal, onConfirm: () => resolve(true), { onClose: () => resolve(false) })
);

const isConfirmed = await confirmModal({question: "Are you sure?"});
```

OFC I can add an event handler to `Cancel` button or corner `X` button, but what about closing stuff programmatically from another place (eg. by clicking backdrop)?

**Comment by hairyf** [comment]
You can use [@overlays/svelte](https://unoverlays.vercel.app/en/core/svelte) as an alternative, which allows for [external control of modals](https://unoverlays.vercel.app/en/core/guide/external-control.html).

**Comment by mattjennings** [comment]
included in v2.0.0
